pkg/k8s/watchers/cilium_network_policy.go

```diff
@@ -160,22 +244,24 @@ func (k *K8sWatcher) onUpsertCNP(
 	// See https://github.com/cilium/cilium/blob/27fee207f5422c95479422162e9ea0d2f2b6c770/pkg/policy/api/ingress.go#L112-L134
 	cnpCpy := cnp.DeepCopy()
 
+	translatedCNP := resolveCIDRGroupRef(cnp, cidrGroupCache)
```

Comment:
```suggestion
	translatedCNP := resolveCIDRGroupRef(cnpCpy, cidrGroupCache)
```

This function should use the copy and that way, it doesn't have to make a copy inside the function.

Reply:
I made it this way because I need to store a copy of the cnp in the cache ahead of the translation. So, in case of a cidr group ref being present, I need to make another copy (the previous one will be stored in the cache and the translated one will be used by add/update cnp code)

Reply:
But I agree that it should use `cnpCpy` anyway, that it's a copy-paste error.